vtkext/private/module/vtkF3DRenderer.cxx

```diff
@@ -655,6 +687,110 @@ void vtkF3DRenderer::ConfigureGridUsingCurrentActors()
   this->ResetCameraClippingRange();
 }
 
+//----------------------------------------------------------------------------
+vtkBoundingBox vtkF3DRenderer::ComputeVisiblePropOrientedBounds(const vtkMatrix4x4* matrix)
+{
+  const auto isMatrixAxisAligned = [](const vtkMatrix4x4* m, const double tol = 1e-8)
+  {
+    for (size_t i = 0; i < 3; ++i)
+    {
+      size_t nonzerosI = 0;
+      size_t nonzerosJ = 0;
+      for (size_t j = 0; j < 3; ++j)
+      {
+        if (::abs(m->Element[i][j]) > tol)
+        {
+          nonzerosI++;
+        }
+        if (::abs(m->Element[j][i]) > tol)
+        {
+          nonzerosJ++;
+        }
+      }
+      if (nonzerosI > 1 || nonzerosJ > 1)
+      {
+        return false;
+      }
+    }
+    return true;
+  };
+
+  /* Use `PokeMatrix` around the call to `GetBounds()` to extend box.
+   * Only gives the thightest bounds if the transformation is axis-aligned. */
+  const auto extendBoxAxisAligned = [&](vtkProp3D* prop3d, vtkBoundingBox& box)
+  {
+    vtkNew<vtkMatrix4x4> tmpMatrix;
+    vtkMatrix4x4::Multiply4x4(matrix, prop3d->GetMatrix(), tmpMatrix);
+    prop3d->PokeMatrix(tmpMatrix);
+
+    box.AddBounds(prop3d->GetBounds());
+
+    prop3d->PokeMatrix(nullptr);
+  };
+
+  /* Use custom logic to extend box.
+   * Should give the tightest bounds even when non-axis-aligned.
+   * Only implemented for `vtkPolyDataMapper`-based actors. */
```

Comment:
should that computation be added to VTK ? It looks useful.

Reply:
Not a VTK expert myself so I'll let you ponder that one ;p